### PR TITLE
feature: 1750 switch from curl to guzzle

### PR DESCRIPTION
**Pull request by alli83**
# Pull request for issue: #1750 

This is a pull request for the following functionalities:

switch from curl to guzzle

## How to test?

connect as admin
go to Dataset
e.g update a dataset
click mint DOI

## Any changes to documentation?

Describe changes to the documentation

## Any improvements to CI/CD pipeline?

Describe any improvements to the Gitlab pipeline

### CODE REVIEW

File: protected/views/adminDataset/_form.php [hunk 1]
```diff
@@ -358,15 +358,15 @@
...
                                                 } else if (output.create_md_status == 201 && output.create_doi_status == 201) {
                                                     $("#minting").addClass("alert alert-success").html("New DOI successfully minted");
-                                                } else if (output.check_metadata_status == 404 && output.check_doi_status == 404 && output.create_md_status == 422 && output.create_doi_status == 422) {
+                                                } else if (&& output.check_doi_status == 404 && output.create_md_status == 422 && output.create_doi_status == 422) {
                                                     $("#minting").addClass("alert alert-danger").html("This DOI cannot be created because of the metadata status: " + output.create_md_status + ", and the doi status: " + output.create_doi_status + " Details can be found at <a href=\'https://support.datacite.org/reference/mds#api-response-codes\' target=\'_blank\'>here</a>");
```
Comment (alli83) [inline protected/views/adminDataset/_form.php]: I don't think we should have the message from the response

**Review by rija — COMMENTED** [review]
Hi @alli83,

**praise**: code change looks fine
**praise**: great to see yii2 DI container being used for loading the Guzzle client
**chore**: Add CHANGELOG entry
**issue**: Acceptance tests (the ones related to the mint doi workflow) are failing in `tests/acceptance/AdminDatasetForm.feature`
**question**: What do you mean by "I don't think we should have the message from the response" in your comment about [protected/views/adminDataset/_form.php](https://github.com/gigascience/gigadb-website/pull/2037/files#diff-c70744b356df2a7d1163442f1e1efff7c89740ea1746c96cdeae054eaa0e3041)
chore: rebase so you get the fix to the PostgreSQL error

**Comment by pli888** [comment]
@alli83 Can you rebase with the latest code from develop please? This will pick up the fix to the acceptance tests in `AdminDatasetForm.feature`.

**Review by pli888 — COMMENTED** [review]
**praise**: I was able to create a new DOI for a new dataset and update metadata for an existing dataset using the Datacite test API from my dev environment.
**praise**: Code looks good to me. It has been significantly simplified in functional test class and AdminDatasetController with the  dependency injection container pattern for using Guzzle.
**praise**: All tests are passing in my CI pipeline.
**suggestion**: Can you take a look at the `protected/components/DownloadService` class please? I think it can do with updating with the dependency injection container way of using Guzzle.

**Review by rija — COMMENTED** [review]
Hi @alli83,

**praise**: automated tests are passing locally and on CI
**chore**: as mentioned by @pli888 already, there are couple of files in conflicts, I suggest rebasing onto develop's HEAD

**issue**: When I deploy to AWS using Gitlab pipeline, I've got 500 server error when I click "Mint DOI" (which doesn't happen on the main branch)
`Failed to load resource: the server responded with a status of 500 (Internal Server Error)` 
the docker container log has more info and seems to indicate an issue with guzzle http client and the DI container configuration:
`ReflectionException: Class guzzleHttpClient does not exist in /var/www/vendor/yiisoft/yii2/di/Container.php:507`
**suggestion**: it sounds like the DI container configuration was not added to the config file used on AWS deployments: `ops/configuration/yii-conf/web.staging.live.php.dist`
**note**: I agree with @pli888's comment on updating `protected/components/DownloadService.php` to use the same approach of using Guzzle

**Review by rija — APPROVED** [review]
Hi @alli83,

**praise**: All tests are passing locally and on CI
**praise**: Mint DOI functionality works fine locally and on my AWS deployment with clear messages shown to the users
**praise**: Great to see DI container-configured Guzzle instance used throughout the codebase

Assuming it is going to be rebased onto latest develop, 
I'm happy to approve this PR

**Review by pli888 — APPROVED** [review]
**praise**: DOI functionality is working on my staging server.
**praise**: Code looks good to me.
**praise**: All tests on my CI pipeline are passing.

Happy to approve this PR.

**Pull request merged**